File: 0506-relative-ranks/0506-relative-ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        def rank(p):
            if(p == 1):
                return "Gold Medal"
            
            if(p == 2):
                return "Silver Medal"
            
            if(p == 3):
                return "Bronze Medal"
            
            return str(p)
        
        n = len(score)
        values = [(s, i) for i, s in enumerate(score)]
        values.sort(reverse=True)
        res = [""] * n
        
        for i, (_, idx) in enumerate(values):
            res[idx] = rank(i + 1)
        
        return res
            
            

File: 0506-relative-ranks/0506-relative-ranks.go
func findRelativeRanks(score []int) []string {
    posMap := make(map[int]int)
    for i, s := range score {
        posMap[s] = i
    }
    
    slices.SortFunc(score, func(a, b int) int { 
        return b - a
    })
    ans := make([]string, len(score))
    for i, s := range score {
        idx := posMap[s]
        switch i {
            case 0:
                ans[idx] = "Gold Medal"
            case 1:
                ans[idx] = "Silver Medal"
            case 2:
                ans[idx] = "Bronze Medal"
            default:
                ans[idx] = strconv.Itoa(i + 1)
        }
    }
    return ans
}